redbot/cogs/cleanup/cleanup.py

```diff
@@ -271,6 +278,48 @@ def check(m):
 
         await mass_purge(to_delete, channel)
 
+    @cleanup.command()
+    @commands.guild_only()
+    @commands.bot_has_permissions(manage_messages=True)
+    async def between(
+        self,
+        ctx: commands.Context,
+        one: RawMessageIds,
+        two: RawMessageIds,
+        delete_pinned: bool = False,
+    ):
+        """Delete the messages between Messsage One and Message Two, providing the messages IDs.
+
+        The first message ID should be the older message, and the second one the newer.
```

Comment:
We could remove the specific requirement of order here by just ordering them by which one was created first, but I actually like that this may help prevent mistakes.